Add API to tell if user code is in active IRQn

AlessandroA opened the pull request:
This introduces a new user API:

`int vIRQ_GetLevel(void)`

that returns:
- `-1`  if no interrupt is active
- `>= 0` if an interrupt is active and it is being served;
       the actual return value indicates the priority level of the IRQn
       as set by default or with `vIRQ_SetPriority(IRQn_Type irqn)`

miklis commented:
Build finished. No test results found.

AlessandroA commented:
@meriac 
@bremoran please tell me if this implements what you need, and if it solves https://github.com/ARMmbed/uvisor/issues/34

AlessandroA commented:
@hugovincent adding you as you were involved in the original discussion. Feel free to comment

meriac commented on core/mbed/source/interrupts.cpp:
```diff
@@ -106,3 +106,38 @@ uint32_t vIRQ_GetPriority(uint32_t irqn)
+    if (__uvisor_mode == 0) {
+        /* check if an IRQn is active (an ISR is being served) */
+        uint32_t ipsr = __get_IPSR();
+        int irqn = (int) (ipsr & 0x1FF) - UVISOR_NVIC_IRQn_OFFSET;
```
Ensure that mask is appropriate for selected target. The NVIC->IABR register (as use in NVIC_GetActive) is only 8 DWORD wide - which leaves only 256 interrupts. 0x1FF would reult in a overflow.

meriac commented on core/system/src/unvic.c:
```diff
@@ -196,6 +196,31 @@ uint32_t unvic_irq_priority_get(uint32_t irqn)
+    /* gather IPSR from exception stack frame */
+    /* the currently active IRQn is the one of the SVCall, while instead we want
+     * to know the IRQn at the time of the SVCcall, which is saved on the stack */
+    uint32_t ipsr = box_sp[7];
```
please drop svc_cx_validate_sf and use unprivileged read instead

meriac commented on core/system/src/unvic.c:
```diff
@@ -196,6 +196,31 @@ uint32_t unvic_irq_priority_get(uint32_t irqn)
+    /* the currently active IRQn is the one of the SVCall, while instead we want
+     * to know the IRQn at the time of the SVCcall, which is saved on the stack */
+    uint32_t ipsr = box_sp[7];
+    uint32_t irqn = (ipsr & 0x1FF) - IRQn_OFFSET;
```
0x1FF results in an overflow of NVIC_GetActive

miklis commented:
Build finished. No test results found.

AlessandroA commented:
@meriac 

- removed use of `svc_cx_validate_sf()`
- added check on `irqn` (fail if `>= HW_INTERRUPTS`)
    - note: this check is included in `unvic_check_default()` in uVisor, and is added explicitly in the mbed-facing fallback version (for when uVisor is disabled)
